exec/java-exec/src/main/java/org/apache/drill/exec/physical/rowSet/RowSetWriterImpl.java

```diff
@@ -129,7 +129,7 @@ public RowSetWriter addRow(Object...values) {
 
   @Override
   public RowSetWriter addSingleCol(Object value) {
-    return addRow(new Object[] {value});
+    return addRow(value);
```

Comment:
Won't work. Must be an array. The idea is that addRow() takes an array of columns. But, if the row has a single column, and that one column is repeated, the Java code is ambiguous. The original (correct) implementation handles this case.

Reply:
While this would seem to be true, it won't work for the case of a map:

addRow(new Object[]{"foo", 10})

Is the above two column values, or a single column that happens to be an object?

There is actually no need to debate this, however. Without this feature, some unit tests will fail. (I know that because I added this method directly in response to those test failures...)

Still, if you can get those tests to pass, even with the map value ambiguity, then my concern will be addressed.

Reply:
Tests passed with this change. I'm afraid that we are talking about different cases.
`addRow(new Object[]{"foo", 10})` will work as you expected - it will handle incoming array as two column values, but the code in the method which was changed is different a little bit, it is similar to this call:
`addRow((Object) new Object[]{"foo", 10})` - in this case, it will be handled as a single column with array value.

Reply:
Ah! But I don't want to do the cast; that is the purpose of this method. I want:

```
  ...addRow(mapValue(10, "foo"), 20); // convenience
  ...addRow(new Object[] {10, "foo"}, 20); // low-level
...
  ...addSingleCol(mapValue(10, "foo"); // convenience
 ... addSingleCol(new Object[] {10, "foo"}); // low-level
```

Unless we wrap that single column in another array, the second form is the same as:

```
 ... addRow(new Object[] {10, "foo"});
```

Which will be interpreted as a set of two columns, not as a single column with a map-value.

I could be wrong, and I'd have to test actual code to be sure. In fact, I'll do that test this weekend.

Reply:
No need to cast, it was an example. In the method where the value is used reference type is already `Object`, so we don't need to do a cast.

> Unless we wrap that single column in another array, the second form is the same as:
> ```
>  ... addRow(new Object[] {10, "foo"});
> ```

Actually not, it will be the same as
```
addRow((Object) new Object[] {10, "foo"});
```
because method argument in `addSingleCol` has `Object` type.

Reply:
Thanks for the explanation, makes sense.